src/routes/CreateNewSafePage/CreateNewSafePage.tsx

```diff
@@ -0,0 +1,195 @@
+  const location = useLocation()
+  const safeRandomName = useMnemonicSafeName()
+
+  async function showSafeCreationProcess(newSafeFormValues: CreateSafeFormValues) {
+    await saveToStorage(SAFE_PENDING_CREATION_STORAGE_KEY, { ...newSafeFormValues })
+    setSafePendingToBeCreated(newSafeFormValues)
+  }
+
+  const [initialFormValues, setInitialFormValues] = useState<CreateSafeFormValues>()
+
+  useEffect(() => {
+    if (provider && userWalletAddress) {
+      const initialValuesFromUrl = getInitialValues(userWalletAddress, addressBook, location, safeRandomName)
+      setInitialFormValues(initialValuesFromUrl)
+    }
+  }, [provider, userWalletAddress, addressBook, location, safeRandomName])
+
+  const isProductionEnv = APP_ENV === 'production'
+
+  if (isLoading) {
+    return (
+      <div className={classes.loaderContainer} data-testid={'create-new-safe-loader'}>
+        <Loader size="md" />
+      </div>
+    )
+  }
+
+  return !!safePendingToBeCreated ? (
+    <SafeCreationProcess />
+  ) : (
+    <Page>
+      <Block>
+        <Row align="center">
+          <IconButton disableRipple onClick={history.goBack} className={classes.backIcon}>
+            <ChevronLeft />
+          </IconButton>
+          <Heading tag="h2">Create new Safe</Heading>
+        </Row>
+        <StepperForm
+          initialValues={initialFormValues}
+          onSubmit={showSafeCreationProcess}
+          testId={'create-new-safe-form'}
+        >
+          {!isProductionEnv && (
+            <StepFormElement
+              label={selectWalletAndNetworkStepLabel}
+              nextButtonLabel="Continue"
+              disableNextButton={!provider}
+            >
+              <SelectWalletAndNetworkStep />
+            </StepFormElement>
+          )}
+          <StepFormElement label={nameNewSafeStepLabel} nextButtonLabel="Continue">
+            <NameNewSafeStep />
+          </StepFormElement>
+          <StepFormElement
+            label={ownersAndConfirmationsNewSafeStepLabel}
+            nextButtonLabel="Continue"
+            validate={ownersAndConfirmationsNewSafeStepValidations}
+          >
+            <OwnersAndConfirmationsNewSafeStep />
+          </StepFormElement>
+          <StepFormElement label={reviewNewSafeStepLabel} nextButtonLabel="Create">
+            <ReviewNewSafeStep />
+          </StepFormElement>
+        </StepperForm>
+      </Block>
+    </Page>
+  )
+}
+
+export default CreateNewSafePage
+
+const useStyles = makeStyles((theme) => ({
+  backIcon: {
+    color: theme.palette.secondary.main,
+    padding: sm,
+    marginRight: '5px',
+  },
+  loaderContainer: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    height: '100%',
+  },
+}))
+
+const DEFAULT_THRESHOLD_VALUE = 1
+
+// initial values can be present in the URL because the Old MultiSig migration
+function getInitialValues(userAddress, addressBook, location, suggestedSafeName): CreateSafeFormValues {
```

Comment:
Instead of passing the location object (and having to rely on `query-string`) you could just use the `useParams()` hook from `react-router` and pass the object outputted as an argument instead. You could then uninstall `query-string` too if you installed it for this ticket?